drf_kit/cache.py

```diff
@@ -38,6 +38,25 @@ class BodyCacheKeyConstructor(CacheKeyConstructor):
 
 
 class CacheResponse(decorators.CacheResponse):
+    def __init__(self, timeout=None, key_func=None, cache=None, cache_errors=None, cache_error_func=None):
+        """
+        :param cache_error_func: Function to be called when an error occurs while trying to get the cache.
```

Comment:
Thinking about the general DX of this solution here. First, I only partially agree with what was stated in the issue description:

> @cache_response decorator is used to apply cache to view sets. Usually, attempts to fetch cached data is a "nice to have" to any view call in terms of performance. Therefore, any error on cache should not fail the view call. Instead, a more resilient approach is to ignore cache on errors, and proceed to usual view call.

Attempts to fetch cached data are, indeed, **usually** "nice to have", but emphasis on **usually**: for an endpoint that is heavily accessed and has a cache hit ratio well over 90%, it is not a nice to have, it is part of the infrastructure. If the cache fails and we send all requests upstream as if there was no cache, we're likely to cause a self-inflicted DDoS since the load will increase drastically in a moment's notice. If this unexpected load top up database capacity, we might have a bigger (broader) availability problem than just the errors on the cached endpoint.

In that sense, while we could have a default behavior of calling the view, perhaps this should be configurable on specific endpoints in which this is not a real option (especially considering drf_kit is not Nilo-specific). Since we take a `cache_error_func`, this is actually already possible (all that is needed is re-raising the exception on `cache_error_func`), so perhaps there is nothing needed to address the "GET" case.

As for the "SET" case, it is a bit more complicated: besides the same considerations done for the "GET" case, we need to consider that there are two possible scenarios:

- There is no cached data for the endpoint. In this case, not setting the cache hurts performance, but doesn't hurt correctness.
- There is already cached data for the endpoint. In this case, not setting the cache doesn't hurt performance (as future GET requests will hit the cache), but it hurts correctness (a stale value will be read)

So I don't think ignoring cache fails for updates should be the default behavior since it introduces data inconsistencies (but I'm okay with apps that use drf kit being able to opt-in for this behavior)

Reply:
Good point! I was thinking over of whether cache was indeed a "nice to have", and you make it clear that it **might** be, but might not be as well.

Just to make clear, your suggestion is to make it disabled default , and the user has to opt-in to have their error ignored? If it is, sounds good to me. The set method can be something like:

```
        try:
            self.cache.set(key, response_dict, self.timeout)
        except Exception as exc:
            if self.cache_error_func:
                self.cache_error_func(exception=exc, key=key, request=request, get_cache=False)
            else:
                raise from exc
```